docs/maintaining: add deprecation policy (#3866)
nix-community/nixvim: MAINTAINING.md

## Deprecation

We aim to guarantee a seamless experience for stable channel users.

Deprecations should include a comment in the format of:
```nix
  # TODO: Added YYYY-MM-DD, Remove after <n+2>
```

where:

- n is the current release (i.e. 2025.05 as of writing)

> - n+1 is the next release (i.e. 2025.11 as of writing)

Reviewer: ```suggestion
- `n` is the current release; 25.05 at the time of writing
- `n+1` is the next release; 25.11 at the time of writing
```

I'm not keen on having the actual current release written down; it's yet another thing to update when we cut a new release. Although we already have to grep the codebase for `s/25.05/25.11/` so it doesn't make things much worse.

We should use the actual naming convention though, `YY.MM` instead of `YYYY.MM`.

Would it make more sense to mention `n+2` (26.05), since that is the release we actually care about? Mentioning `n+1` here could get confusing 😵‍💫

Author: > it's yet another thing to update when we cut a new release

I don't think this needs to be updated. It's okay for this to be out of date as we clarify that this was at the time of writing. It just serves as a practical example for what we mean by these values. I think not mentioning the values might be a bit too abstract and difficult to understand.

If a warning is created on release n, then:

- We **should not** (unless there is a bug, or it is plain wrong) remove it before n+1 is released

> - We **might** remove it if it annoys us starting from when n+1 is released

Reviewer: ```suggestion
- We **may** remove it after n+1 is released, but only if it cases undue friction
```
